### PR TITLE
Implement SMB share settings datasource

### PR DESCRIPTION
**Pull request by sapana05**
# Description
Implement SMB share settings datasource

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Feature Pull Request

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by sapana05** [comment]
image
image
image
image

**Comment by sapana05** [comment]
run e2e test

File: examples/data-sources/powerscale_smb_share_settings/provider.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+*/
+terraform {
+  required_providers {
+    powerscale = {
+      source = "registry.terraform.io/dell/powerscale"
+    }
+  }
+}
+
+provider "powerscale" {
```
Comment (RayLiu7) [inline examples/data-sources/powerscale_smb_share_settings/provider.tf:25]: Don't forget to change it back to default config

File: powerscale/provider/smb_share_settings_data_source.go [hunk 2]
```diff
@@ -0,0 +1,378 @@
+/*
+Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (RayLiu7) [inline powerscale/provider/smb_share_settings_data_source.go:2]: The copyright year should be 2024

File: powerscale/provider/smb_share_settings_data_source.go [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+		zoneStr = filter.Zone.ValueString()
+	}
+	if zoneStr == "" {
+		zoneStr = "System"
+	}
+	idValue := "smb_share_settings_" + zoneStr
+
+	state.SmbShareSettings = &settings
+	state.ID = types.StringValue(idValue)
+	//state.SmbShareSettings.Zone = types.StringValue(zoneStr)
```
Comment (RayLiu7) [inline powerscale/provider/smb_share_settings_data_source.go:367]: redundant line here

File: examples/data-sources/powerscale_smb_share_settings/provider.tf (hunk 1 above)
Comment (sapana05) [reply examples/data-sources/powerscale_smb_share_settings/provider.tf:25]: changed default config

File: powerscale/provider/smb_share_settings_data_source.go (hunk 2 above)
Comment (sapana05) [reply powerscale/provider/smb_share_settings_data_source.go]: updated copyright year

File: powerscale/provider/smb_share_settings_data_source.go (hunk 3 above)
Comment (sapana05) [reply powerscale/provider/smb_share_settings_data_source.go]: removed commented line from here

**Comment by sapana05** [comment]
run e2e test

**Review by RayLiu7 — APPROVED** [review]

**Comment by sapana05** [comment]
run e2e test

**Pull request merged**